.changeset/tiny-apples-wait.md

```diff
@@ -0,0 +1,10 @@
+---
+"@chakra-ui/pin-input": minor
+---
+
+Previously isComplete was being determined based off of old state values as it
+was determined before the setState took effect. This caused the last pin-input
+entry to not call onComplete when updated. Updating the isComplete check to
+reference the next values (the value of the state after rerender) allows the
+check to use the most up to date values in the pin-input fields fixing the last
+digit update issue.
```

Comment:
Always appreciate lots of detail but this is more appropriate for the PR description than the release notes. Better to just let users know what was fixed, rather than what was wrong and how it was fixed.

```suggestion
Resolved an issue where completing character entry in `PinInput` failed to call `onComplete`.
```

Reply:
Thanks for the suggestion! Updated.

Reply:
It's all good! Thanks for taking the time to fix this 😄